portality/view/oaipmh.py

```diff
@@ -409,51 +420,57 @@ def _parameterised_list_identifiers(dao, base_url, specified_oai_endpoint, metad
         #    datetime.strptime(until_date, "%Y-%m-%d")
     #except:
     #    return BadArgument(base_url)
-    
+
     # get the result set size
     list_size = app.config.get("OAIPMH_LIST_IDENTIFIERS_PAGE_SIZE", 25)
-    
+
     # decode the oai_set to something we can query with
     try:
         decoded_set = decode_set_spec(oai_set) if oai_set is not None else None
     except SetSpecException:
         return BadArgument(base_url)
-    
+
     for f in formats:
         if f.get("metadataPrefix") == metadata_prefix:
             # do the query and set up the response object
-            total, results = dao.list_records(from_date, until_date, decoded_set, list_size, start_number)
-            
+            total, results = dao.list_records(from_date, until_date, decoded_set, list_size, start_number, start_after)
+
             # if there are no results, PMH requires us to throw an error
             if len(results) == 0:
                 return NoRecordsMatch(base_url)
-            
+
+            # get the full total
+            full_total = total
+            if start_after is not None:
+                full_total = total + start_number - start_after[1]
```

Comment:
I didn't understand this bit of code, sorry.  start_number seems to have become redundant elsewhere, so I'm not sure how it is being used here, or how full_total is different to total.  Can you just add some comments to explain?

Reply:
In the xml for the resumption token, it has complete_list_size and cursor

Since we are doing new searches each time, the total will be a reduced total, not the same as the one received in the first search. So I recalculate the total, and return the complete list size.

We would need to hold onto the start number so we can return the current cursor for the resumption token xml

Reply:
@richard-jones comment added to code and code modified to fix bug you spotted

Reply:
Ah, got it!  Clever!